cirq/vis/heatmap.py

```diff
@@ -52,220 +36,328 @@ def relative_luminance(color: np.ndarray) -> float:
+            text_kwargs.update(self._config.get('annotation_text_kwargs', {}))
+            ax.text(x, y, annotation, **text_kwargs)
+
+    def _plot_on_axis(self, ax: plt.Axes, **collection_options: Any) -> mpl_collections.Collection:
+        # Step-1: Convert value_map to a list of polygons to plot.
+        polygon_list = self._get_polygon_units()
+        collection: mpl_collections.Collection = mcoll.PolyCollection(
+            [c.polygon for c in polygon_list], cmap=self._config['colormap'], **collection_options
+        )
+        collection.set_clim(self._config.get('vmin'), self._config.get('vmax'))
+        collection.set_array(np.array([c.value for c in polygon_list]))
+        # Step-2: Plot the polygons
+        ax.add_collection(collection)
+        collection.update_scalarmappable()
+        # Step-3: Write annotation texts
+        if self._config.get('annotation_map') or self._config.get('annotation_format'):
+            self._write_annotations([(c.center, c.annot) for c in polygon_list], collection, ax)
+        ax.set(xlabel='column', ylabel='row')
+        # Step-4: Draw colorbar if applicable
+        if self._config.get('plot_colorbar'):
+            self._plot_colorbar(collection, ax)
+        # Step-5: Set min/max limits of x/y axis on the plot.
+        rows = set([q.row for qubits in self._value_map.keys() for q in qubits])
+        cols = set([q.col for qubits in self._value_map.keys() for q in qubits])
+        min_row, max_row = min(rows), max(rows)
+        min_col, max_col = min(cols), max(cols)
+        min_xtick = np.floor(min_col)
+        max_xtick = np.ceil(max_col)
+        ax.set_xticks(np.arange(min_xtick, max_xtick + 1))
+        min_ytick = np.floor(min_row)
+        max_ytick = np.ceil(max_row)
+        ax.set_yticks(np.arange(min_ytick, max_ytick + 1))
+        ax.set_xlim((min_xtick - 0.6, max_xtick + 0.6))
+        ax.set_ylim((max_ytick + 0.6, min_ytick - 0.6))
+        # Step-6: Set title
+        if self._config.get("title"):
+            ax.set_title(self._config["title"], fontweight='bold')
+        return collection
+
+    def plot(
+        self, ax: Optional[plt.Axes] = None, **collection_options: Any
+    ) -> Tuple[plt.Axes, mpl_collections.Collection]:
+        """Plots the heatmap on the given Axes.
+        Args:
+            ax: the Axes to plot on. If not given, a new figure is created,
+                plotted on, and shown.
+            collection_options: keyword arguments passed to mcoll.PolyCollection().
+        Returns:
+            A 2-tuple ``(ax, collection)``. ``ax`` is the `plt.Axes` that
+            is plotted on. ``collection`` is the collection of paths drawn and filled.
+        """
+        show_plot = not ax
+        if not ax:
+            fig, ax = plt.subplots(figsize=(8, 8))
+        collection = self._plot_on_axis(ax, **collection_options)
+        if show_plot:
+            fig.show()
+        return (ax, collection)
+
+
+class TwoQubitHeatmap(Heatmap):
```

Comment:
How about naming it with the interaction in it? 
```suggestion
class TwoQubitInteractionHeatmap(Heatmap):
```

Reply:
Or `QubitPairsHeatmap` or `CouplerHeatmap` though we don't really use the concept of a coupler here.

Reply:
done